.github/workflows/package.yml

```diff
@@ -7,33 +7,9 @@ on:
     - ./pom.xml
 
 jobs:
-  tests:
+  tests_and_build:
```

Comment:
Don't do composite action if you don't have multiple use cases. Execute tests and build only on feature branches. (main is obviously good)